fluentd/logger/log.rb

```diff
@@ -0,0 +1,384 @@
+      time, msg = event(:trace, args)
+      puts [@color_trace, caller_line(time, @depth_offset, LEVEL_TRACE), msg, @color_reset].join
+    rescue
+      # logger should not raise an exception. This rescue prevents unexpected behaviour.
+    end
+    alias TRACE trace
+
+    def trace_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_TRACE)
+    end
+
+    def on_debug(&block)
+      return if @level > LEVEL_DEBUG
+      block.call if block
+    end
+
+    def debug(*args, &block)
+      return if @level > LEVEL_DEBUG
+      args << block.call if block
+      time, msg = event(:debug, args)
+      puts [@color_debug, caller_line(time, @depth_offset, LEVEL_DEBUG), msg, @color_reset].join
+    rescue
+    end
+    alias DEBUG debug
+
+    def debug_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_DEBUG)
+    end
+
+    def on_info(&block)
+      return if @level > LEVEL_INFO
+      block.call if block
+    end
+
+    def info(*args, &block)
+      return if @level > LEVEL_INFO
+      args << block.call if block
+      time, msg = event(:info, args)
+      puts [@color_info, caller_line(time, @depth_offset, LEVEL_INFO), msg, @color_reset].join
+    rescue
+    end
+    alias INFO info
+
+    def info_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_INFO)
+    end
+
+    def on_warn(&block)
+      return if @level > LEVEL_WARN
+      block.call if block
+    end
+
+    def warn(*args, &block)
+      return if @level > LEVEL_WARN
+      args << block.call if block
+      time, msg = event(:warn, args)
+      puts [@color_warn, caller_line(time, @depth_offset, LEVEL_WARN), msg, @color_reset].join
+    rescue
+    end
+    alias WARN warn
+
+    def warn_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_WARN)
+    end
+
+    def on_error(&block)
+      return if @level > LEVEL_ERROR
+      block.call if block
+    end
+
+    def error(*args, &block)
+      return if @level > LEVEL_ERROR
+      args << block.call if block
+      time, msg = event(:error, args)
+      puts [@color_error, caller_line(time, @depth_offset, LEVEL_ERROR), msg, @color_reset].join
+    rescue
+    end
+    alias ERROR error
+
+    def error_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_ERROR)
+    end
+
+    def on_fatal(&block)
+      return if @level > LEVEL_FATAL
+      block.call if block
+    end
+
+    def fatal(*args, &block)
+      return if @level > LEVEL_FATAL
+      args << block.call if block
+      time, msg = event(:fatal, args)
+      puts [@color_fatal, caller_line(time, @depth_offset, LEVEL_FATAL), msg, @color_reset].join
+    rescue
+    end
+    alias FATAL fatal
+
+    def fatal_backtrace(backtrace=$!.backtrace)
+      dump_stacktrace(backtrace, LEVEL_FATAL)
+    end
+
+    def puts(msg)
+      if $openShiftLogger.nil?  
+        @out.puts(msg)
+        @out.flush
+        msg
+      else
+        $openShiftLogger.info("#{msg}")
```

Comment:
`$openShiftLogger.info(msg)`
no need to convert `msg` to a string - it should already be a string

Reply:
https://ruby-doc.org/stdlib-2.0.0/libdoc/logger/rdoc/Logger.html
I think you should use the `<<` method - it looks like the message is already formatted - not sure.  If the message is not formatted, then you'll need to configure the formatter to format the messages exactly like fluentd does.  Same with `write` below.